src/Collection/TasksCollection.php

```diff
@@ -39,6 +39,21 @@ public function filterByTestSuite(TestSuiteInterface $testSuite = null)
         });
     }
 
+    /**
+     * @param string[] $tasks
+     *
+     * @return TasksCollection
+     */
+    public function filterByTaskName($tasks)
+    {
+        return $this->filter(function (TaskInterface $task) use ($tasks) {
+            if (empty($tasks)) {
```

Comment:
No need to iterate through the list when the tasks are empty. You can put the empty check above the filter function and return a new collection through `return new TasksCollection($this->toArray());`
See `\GrumPHP\Collection\TasksCollection::filterByTestSuite()`